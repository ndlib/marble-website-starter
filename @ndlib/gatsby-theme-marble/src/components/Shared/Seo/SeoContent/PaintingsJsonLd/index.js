import React from 'react'
import PropTypes from 'prop-types'
import { JsonLd } from '../JsonLd'

export default function PaintingsJsonLd ({ title, description, date, image, author, creditText, dimensions, thumbnail, keywords }) {
  return (
    <JsonLd>
      {{
        '@context': 'https://schema.org',
        '@type': 'VisualArtwork',
        'name': title,
        'dateCreated': date,
        'description': description,
        'image': image,
        'thumbnailUrl': thumbnail,
        'keywords': keywords,
        'size': dimensions,
        'artform': 'painting',
        'creator': [
          {
            '@type': 'Person',
            'name': author,
          },
        ],
        'artMedium': 'canvas',
        'creditText': creditText,
      }}
    </JsonLd>
  )
}

PaintingsJsonLd.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  date: PropTypes.string,
  image: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  creditText: PropTypes.string,
  dimensions: PropTypes.string,
  thumbnail: PropTypes.string.isRequired,
  keywords: PropTypes.string,
}
