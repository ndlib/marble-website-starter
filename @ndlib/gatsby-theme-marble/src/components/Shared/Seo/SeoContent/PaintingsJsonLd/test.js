import React from 'react'
import renderer from 'react-test-renderer'
import PaintingsJsonLd from './'

describe('PaintingsLd component', () => {

  it('renders props into values', () => {
    const tree = renderer
      .create(<PaintingsJsonLd
        title={'Fake Title'}
        description={'This is a fake description'}
        image={'http://fake.io/image'}
        author={'Fake Author'}
        creditText={'Fake Creditor'}
        dimensions={'2 x 4'}
        thumbnail={'http://fake.io/thumbnail'} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
