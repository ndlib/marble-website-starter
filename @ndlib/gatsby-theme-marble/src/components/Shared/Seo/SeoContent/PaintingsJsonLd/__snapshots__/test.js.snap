// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaintingsLd component renders props into values 1`] = `null`;
